tests/ui/needless_pass_by_value.rs

```diff
@@ -5,7 +5,8 @@
     clippy::redundant_pattern_matching,
     clippy::many_single_char_names,
     clippy::option_option,
-    clippy::redundant_clone
+    clippy::redundant_clone,
+    clippy::uninit_assumed_init
```

Comment:
Have to allow `uninit_assumed_lint` here because one of the `needless_pass_by_values` test cases has undefined behavior that was going uncaught till now.